### DISCUSSION TITLE
Bug in Cypher queries when trying to return path

### DISCUSSION THREAD

**Discussion by datanostra** [ArcadeData/arcadedb#369] [Q&A] [answered]
_ArcadeDB version : 22.1.3_

When I'm performing simple path finding Cypher queries, I would like to be able to return all objects present in the paths, but I can't
can't get my paths in the response.

This query returns an error:
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN path`
```
Caused by: java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.util.Map (java.util.ArrayList and java.util.Map are in module java.base of loader 'bootstrap')
        at com.arcadedb.gremlin.query.CypherQueryEngine.transformMap(CypherQueryEngine.java:91)
        at org.apache.tinkerpop.gremlin.arcadedb.structure.ArcadeCypher.execute(ArcadeCypher.java:71)
        at com.arcadedb.gremlin.query.CypherQueryEngine.command(CypherQueryEngine.java:66)
```

I have also tried to collect my objects:
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN a,b,r`
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN nodes(path)`
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN [n in nodes(path) | n.id]`

But all these queries return en empty result:
`{"result":{"vertices":[],"records":[],"edges":[]},"user":"root","version":"22.1.3-SNAPSHOT (build d38e26aa6978b19d0d69999919b5c076fc994d37/1642118408201/main)"}`

The only thing I can do for now is returning simple objects:
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN a`
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN b`
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN r`

**Comment by datanostra** [comment]
This query seems to work with version 22.2.1:
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN a,b,r`

Need to set "serializer" : "graph" in the query.

Using this we are able to collect vertices and edges.

However it is still having the same error when working with the whole path: 
`MATCH path = (a:Transaction{id:'node_a'})-[r:IS_IN]-(b:City{id:'node_b'}) RETURN path`

**Comment by lvca** [comment]
@datanostra do you have a quick script in CYPHER I can run on an empty database to populate it and check your queries?

**Comment by datanostra** [comment]
```
CREATE (n:Transaction {id:'T1'}) RETURN n
CREATE (n:City {id:'C1'}) RETURN n
MATCH
  (t:Transaction),
  (c:City)
WHERE t.id = 'T1' AND c.id = 'C1'
CREATE path = (t)-[r:IS_IN]->(c)
RETURN type(r)
```

Then, you should get the error:
`MATCH path = (t:City{id:'C1'})-[r]-(c:Transaction{id:'T1'}) RETURN path`

**Comment by lvca** [comment, marked as answer]
Fixed in https://github.com/ArcadeData/arcadedb/issues/383.
